renku/service/views/__init__.py

```diff
@@ -0,0 +1,96 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright 2019 - Swiss Data Science Center (SDSC)
+# A partnership between École Polytechnique Fédérale de Lausanne (EPFL) and
+# Eidgenössische Technische Hochschule Zürich (ETHZ).
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Renku service views."""
+import os
+import re
+from functools import wraps
+
+import jwt
+from flask import request, jsonify
+from git import Repo
+
+from renku.service.config import ALLOWED_EXTENSIONS, CACHE_PROJECTS_PATH, \
+    INVALID_PARAMS_ERROR_CODE, CACHE_UPLOADS_PATH, JWT_KEY, JWT_ALGORITHM
+
+is_git_url = re.compile(
+    r'((git|ssh|http(s)?)|(git@[\w\.]+))(:(//)?)([\w\.@\:/\-~]+)(\.git)(/)?',
+    flags=re.I
+)
+
+
+def allowed_file(filename):
+    """Check if filename is allowed."""
+    has_extension = '.' in filename
+    is_allowed = filename.rsplit('.', 1)[1].lower() in ALLOWED_EXTENSIONS
+    return has_extension and is_allowed
+
+
+def find_file(user, file_id):
+    """Search for cached file."""
+    for filename in os.listdir(CACHE_UPLOADS_PATH):
+        metadata = jwt.decode(filename, JWT_KEY, algorithms=[JWT_ALGORITHM])
```

Comment:
I don't like the idea of using filenames to store metadata. Why not use a file-based database like SQLite to store file metadata? That'd improve performance as well.

Reply:
In a deployment you would then save the DB on the persistent volume? Why not just use postgres then?

Reply:
I tried to stay as thin as possible before committing to external processes. There was one important design constraint under which I understood we have to stay and that is that the service should be lightweight as possible. Adding the database and whole distributed processing stack is something I would not consider as "staying lightweight". 

Let's discussed this with next steps.